sql27thsep/sql27th sep.txt

```diff
@@ -0,0 +1,224 @@
+
+1. Extract the middle 3 characters from the string 'ABCDEFG'.
+select SUBSTRING('abcdefg',3,3) as 'middleofstring ';
+
+2. From a table 'Employees' with a column 'FullName', write a query to extract the first name (assuming it's always the first word before a space).
+select	substring(full_name,1,charindex(' ',full_name+' ')-1) from Employees;
```

Comment:
We can simply write like this no : CHARINDEX(' ', FullName) , why fullname is appended with space ?

Reply:
like for names with only one word names..there might not be any spaces so...to identify their name itself as a first name adding space at end of all name

Reply:
so it should be filtered in the WHERE clause right.